src/stirling/source_connectors/socket_tracer/tls_trace_bpf_test.cc

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "src/common/base/base.h"
+#include "src/common/exec/exec.h"
+#include "src/common/testing/test_environment.h"
+#include "src/shared/types/column_wrapper.h"
+#include "src/shared/types/types.h"
+#include "src/stirling/source_connectors/socket_tracer/socket_trace_connector.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/container_images/curl_container.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/container_images/nginx_openssl_3_0_8_container.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/protocol_checkers.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/socket_trace_bpf_test_fixture.h"
+#include "src/stirling/testing/common.h"
+
+namespace px {
+namespace stirling {
+
+namespace tls = protocols::tls;
+
+using ::px::stirling::testing::FindRecordIdxMatchesPID;
+using ::px::stirling::testing::GetTargetRecords;
+using ::px::stirling::testing::SocketTraceBPFTestFixture;
+using ::px::stirling::testing::ToRecordVector;
+
+using ::testing::IsTrue;
+using ::testing::SizeIs;
+using ::testing::StrEq;
+using ::testing::UnorderedElementsAre;
+
+struct TraceRecords {
+  std::vector<tls::Record> tls_records;
+  std::vector<std::string> tls_extensions;
+};
+
+class NginxOpenSSL_3_0_8_ContainerWrapper
+    : public ::px::stirling::testing::NginxOpenSSL_3_0_8_Container {
+ public:
+  int32_t PID() const { return NginxWorkerPID(); }
+};
+
+bool Init() {
+  // Make sure TLS tracing is enabled.
+  FLAGS_stirling_enable_tls_tracing = true;
+
+  // We turn off CQL and NATS tracing to give some BPF instructions back for Mux.
```

Comment:
Outdated comment?

Reply:
Good catch. Updated in 37bcf97a2a3a76f21a6234aec34c48e56f7b45ed to reflect that the disabled protocols give room for TLS instructions.